config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LogoutConfigurer.java

```diff
@@ -519,4 +512,38 @@ public void logout(HttpServletRequest request, HttpServletResponse response, Aut
 
 	}
 
+	private static class OpenSaml4LogoutSupportFactory {
```

Comment:
Is this class still necessary? I think it was necessary when it was acting as a classpath guard. However, since you are using reflection now, I don't think this class offers this same benefit.

Reply:
Those classes make the code cleaner by moving the try-catch from the reflective operations to another method